apollo-router/src/spec/query/tests.rs

```diff
@@ -5900,7 +5899,7 @@ fn filtered_defer_fragment() {
         &mut response,
         None,
         Object::new(),
-        &schema,
+        schema.api_schema(),
```

Comment:
ah, nice catch. Making a separate type for the API schema is a great idea to uncover issues like these

Reply:
Yes, renaming Schema::definitions() to supergraph_schema() tries to go in the same direction